validphys2/src/validphys/dataplots.py

```diff
@@ -855,14 +854,33 @@ def plot_smpdf(pdf, dataset, obs_pdf_correlations, mark_threshold:float=0.9):
 
     plotting_var = info.get_xcol(table)
 
-    #TODO: vmin vmax should be global or by figure?
-    vmin,vmax = min(plotting_var), max(plotting_var)
-    if info.x_scale == 'log':
-        norm = mcolors.LogNorm(vmin, vmax)
+    categorical = not np.issubdtype(plotting_var.dtype, np.number)
+    if categorical:
+        # Categorical
+        keys, values = np.unique(plotting_var, return_inverse=True)
+        plotting_var = values
+        nunique = len(keys)
+        if nunique <= len(cm.Set2.colors):
+            cmap = mcolors.ListedColormap(cm.Set2.colors[:nunique])
+        else:
+            cmap = cm.viridis.resample(nunique)
```

Comment:
I guess this will hardly ever be used, but perhaps something like `gist_ncar` would make it easier to match plot and label than `vidris`. See for some continuous colorschemes: https://matplotlib.org/stable/_images/sphx_glr_colormaps_021.png

Reply:
As you say it doesn't matter, but I think all prefer all the many colors being similarly different to having some that are accidentally very close.